src/main/java/de/hysky/skyblocker/skyblock/CompactDamage.java

```diff
@@ -0,0 +1,72 @@
+package de.hysky.skyblocker.skyblock;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import de.hysky.skyblocker.config.SkyblockerConfig;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.skyblock.item.CustomArmorAnimatedDyes;
+import net.minecraft.entity.decoration.ArmorStandEntity;
+import net.minecraft.nbt.NbtHelper;
+import net.minecraft.text.Text;
+
+import java.text.DecimalFormat;
+
+
+public class CompactDamage {
+
+	private CompactDamage() {
+	}
+
+	private static final DecimalFormat DECIMAL_FORMAT = new DecimalFormat(SkyblockerConfigManager.get().general.compactDamage.format);
+
+	public static void compactDamage(ArmorStandEntity entity) {
+		if (!entity.isInvisible() || !entity.hasCustomName() || !entity.isCustomNameVisible() || entity.getFireTicks() != -1 || !entity.shouldHideBasePlate()) return;
+		//Dmg armor stands have no base plate and have a fire time of -1 (just one of these isn't enough to determine if it's a dmg armor stand)
+		//In fact, even this much checking might not be accurate. Needs testing, or just waiting until someone reports it as an issue
+		JsonObject json = JsonParser.parseString(Text.Serialization.toJsonString(entity.getCustomName(), entity.getRegistryManager())).getAsJsonObject();
```

Comment:
wouldn't it be better to take the entity's name as Text and iterate the siblings? it would be a lot faster and more concise as you can avoid all these JSON checks

Reply:
There is a get siblings method which returns an array list of the extra text objects.